code/game/machinery/doors/airlock.dm

```diff
@@ -37,7 +37,16 @@
 /obj/machinery/door/airlock/attack_generic(var/mob/user, var/damage)
 	if(stat & (BROKEN|NOPOWER))
 		if(damage >= 10)
-			if(src.density)
+			if(src.locked || src.welded)
+				visible_message("<span class='danger'>\The [user] begins breaking into \the [src] internals!</span>")
+				if(do_after(user,10 SECONDS,src))
+					src.locked = 0
+					src.welded = 0
+					update_icon()
+					open(1)
+					if(prob(25))
+						src.shock(user, 100)
+			else if(src.density)

```

Comment:
No, it never has. That was the section we discussed about how to make it so simple_mobs could break through inoperative airlocks. It doesn't have an effect on players. attack_generic is a thing bots and simple_animals do.